packages/common/src/Utils.ts

```diff
@@ -95,12 +106,13 @@ export async function getEip712Signature (
       // @ts-ignore
       method = web3.currentProvider.send
     }
-    method.bind(web3.currentProvider)({
-      method: 'eth_signTypedData' + methodSuffix,
+    const paramBlock = {
+      method: `eth_signTypedData${methodSuffix}`,
       params: [senderAddress, dataToSign],
-      from: senderAddress,
+      jsonrpc: '2.0',
```

Comment:
no, wait, this is a problem, how do you not need a 'from' address? What if the provider controls multiple addresses, which is totally part of an RPC spec afaik?

Reply:
The firsr param is the "from" address, see here:
https://eips.ethereum.org/EIPS/eip-712#specification-of-the-eth_signtypeddata-json-rpc

Reply:
Yes that's what I said :)

Reply:
oh, ok then)) good that hardhat checks that